### ISSUE TITLE
enableAutoSessionTracking API

### ISSUE THREAD

**Issue by lhunath** [getsentry/sentry-cocoa#427]
The API for enabling and disabling auto session tracking is odd and different from the existing API, which is confusing.

If we compare to, for instance, `SentryAutoBreadcrumbTrackingIntegration`, the feature is enabled by including it in the integrations, and disabled by omitting it.

Contrary, `SentryAutoSessionTrackingIntegration` is not enabled by including it in the integrations.  In addition to adding the integration, one also needs to enable a specific custom setting in the sentry options.  Furthermore, turning the option off does not in fact turn off auto session tracking again.

**Comment by bruno-garcia** [comment]
It's true having the Integrations list and the flag on Options to be odd. Currently the [Sentry SK unified API](https://docs.sentry.io/development/sdk-dev/unified-api/) is based on adding/removing (usually by type) integrations but at this point we're considering to replace this approach with controling things through flags on SentryOptions. We'll keep this open to get more feedback and iterate though the API before making a decision.

**Comment by philipphofmann** [comment]
As we have now 
* `enableAutoSessionTracking`
* `enableOutOfMemoryTracking`
* `enableAutoPerformanceTracking`
* `enableNetworkTracking`
* `enableSwizzling`

I think we should add `enableAutoBreadcrumbTracking` to `SentryOptions`. Do you agree @brustolin and @bruno-garcia?

**Comment by philipphofmann** [comment]
Please also update the docs after fixing this.

**Comment by kevinrenskers** [comment]
What's the point of the `integrations` and `defaultIntegrations` options? They are stringly-typed which seems kinda bad, but that's besides the point. I don't really understand why there's an array of integrations that you can add or remove stuff into/from, but separate from that there are all these `enableX` options as well.

Like, these are the default integrations:

```
+ (NSArray<NSString *> *)defaultIntegrations
{
    return @[
        @"SentryCrashIntegration",
#if SENTRY_HAS_UIKIT
        @"SentryANRTrackingIntegration", @"SentryScreenshotIntegration",
        @"SentryUIEventTrackingIntegration",
#endif
        @"SentryFramesTrackingIntegration", @"SentryAutoBreadcrumbTrackingIntegration",
        @"SentryAutoSessionTrackingIntegration", @"SentryAppStartTrackingIntegration",
        @"SentryOutOfMemoryTrackingIntegration", @"SentryPerformanceTrackingIntegration",
        @"SentryNetworkTrackingIntegration", @"SentryFileIOTrackingIntegration",
        @"SentryCoreDataTrackingIntegration"
    ];
}
```

But we have these options as well:

```
@property (nonatomic, assign) BOOL enableAutoSessionTracking;
@property (nonatomic, assign) BOOL enableOutOfMemoryTracking;
@property (nonatomic, assign) BOOL enableAutoPerformanceTracking;
@property (nonatomic, assign) BOOL enableUIViewControllerTracking;
@property (nonatomic, assign) BOOL enableUserInteractionTracing;
@property (nonatomic, assign) BOOL enableNetworkTracking;
@property (nonatomic, assign) BOOL enableFileIOTracking;
@property (nonatomic, assign) BOOL enableCoreDataTracking;
@property (nonatomic, assign) BOOL enableAppHangTracking;
```

And all those options have an entry in the integrations as well. So why are these 2 different concepts of enabling things? 

@philipphofmann you said to add a new `enableAutoBreadcrumbTracking` option, and that seems simple enough (just copy how the other options are done), but I really don't understand the difference between these enable options and the integrations. 

I feel that the OP also talked about exactly this confusion, where sometimes you enable a feature by adding it to the integrations, and disable it by removing it from the integrations (which seems to make sense), but most of the times you need to both include it AND set an enable option to true.

Instead of this ticket making everything more logical by choosing one enable method or the other (either add/remove from integrations, or set the enableX option), we're now going in the direction to also make `SentryAutoBreadcrumbTrackingIntegration` have this confusing way, like the others, while in reality it's currently the only integration that seems to make sense to me?

**Comment by philipphofmann** [comment]
That's an excellent summary of the initial problem, @kevinrenskers. On some other platforms, it's more acceptable to enable features by having a list of integration and adding and removing from this list. On Cocoa, especially with ObjC and Swift, we use code completion when enabling and disabling features. That's why we went for the feature flags on the options instead of having the list of integrations. We still need the list of integrations though of installing them when starting the SDK. So, anybody could still use the default list of integrations and add and remove integrations, but I think nobody does this.

TLDR; use the feature flags of the options and add `enableAutoBreadcrumbTracking`

**Comment by kevinrenskers** [comment]
We can't get rid of the array of integration and ONLY use the flags?

**Comment by philipphofmann** [comment]
> We can't get rid of the array of integration and ONLY use the flags?

For now, not, would be a breaking change.
